client/src/test/TestHelper.ts

```diff
@@ -128,44 +128,33 @@ export default class TestHelper {
     }
 
     public static async checkForRunningProcesses(checkJava: boolean, checkBoogie: boolean, checkZ3: boolean): Promise<void> {
-        const options: SpawnOptionsWithoutStdio = {
-            shell: true
-        };
-        const outputs: Output[] = [];
-        // note for pgrep on macOS and linux:
-        // the pgrep command might show up in the list of running commands. Simply using `-f java.*Viper` can thus lead to false
-        // positives as it matches the command line of just this pgrep command. Two different solutions to this problem are
-        // described here: https://serverfault.com/q/367921
-        // either one can use `-f [j]ava.*Viper` or `-f [^]]java.*Viper` to avoid this problem. As we do have a static string to match
-        // against, we have opted for the first variant.
-        // note that we add `|| true` on macOS and Linux because `pgrep` exits with code 1 if no process is found.
-        if (State.isWin) {
-            function getArgs(whereCond: string): string[] {
-                return ['process', 'where', whereCond, 'get', 'ParentProcessId,ProcessId,Name,CommandLine'];
+        const outputMsgs: string[] = [];
+        const allProcesses = await psList();
+
+        function checkProcess(name: string, checkCmd?: (cmd: string) => boolean): void {
+            const processes = allProcesses.filter(proc => proc.name === name);
+            for (const proc of processes) {
+                if (checkCmd && proc.cmd && !checkCmd(proc.cmd.toLowerCase())) {
```

Comment:
It's not clear to me when ``proc.cmd`` might not be set and whether it makes sense to skip the current process only if is set. I would guess that it's usually set for any process that we care about so this doesn't matter?